### DISCUSSION TITLE
Multistage deployment using Argo

### DISCUSSION THREAD

**Discussion by TannerGabriel** [argoproj/argo-cd#6612] [Q&A]
How would you go about deploying different versions of your application (could also be the same version using promotion) in different Kubernetes namespaces with a single Git repository e.g. having a multistage deployment of your application?

**Comment by wtam2018** [comment]
It looks like you will have to create an Argo CD Application for each namespace/stage you are deploying into.   I was hoping ApplicationSet can be used https://argocd-applicationset.readthedocs.io/en/stable/Generators/#git-generator-files  but the target namespace is the folder name which may not work for you.

**Reply by TannerGabriel** [reply]
The folder naming matching the namespace might work for my use case, but is it necessary to provide a cluster config file since I would like to use the same cluster for both options? Another way would be, as you mentioned using multiple namespaces for the stages and syncing only when validation steps of the previous stage are complete.
